### PR TITLE
Work around INF=0 on some shared hosts

### PR DESCRIPTION
**Pull request by icewind1991**
Possible fix for #15030

See https://bugs.php.net/bug.php?id=69188

DavidWiesner can you verify that this fixes the problem

cc @PVince81

### CODE REVIEW

File: lib/private/helper.php [hunk 1]
```diff
@@ -742,7 +742,7 @@ public static function freeSpace($dir) {
 			$freeSpace = max($freeSpace, 0);
 			return $freeSpace;
 		} else {
-			return INF;
+			return (INF > 0)? INF: PHP_INT_MAX; // work around https://bugs.php.net/bug.php?id=69188
```
Comment (PVince81) [inline lib/private/helper.php]: why not always using PHP_INT_MAX ? (or a custom constant)

Comment (icewind1991) [inline lib/private/helper.php]: `INF` is more "correct" so I would prefer to use it when possible

**Comment by PVince81** [comment]
:+1: makes sense

File: lib/private/helper.php (hunk 1 above)
Comment (rullzer) [inline lib/private/helper.php]: Also if you have a correct setup `INF > PHP_INT_MAX`

**Comment by rullzer** [comment]
:+1:

**Pull request merged**